src/main/java/org/folio/config/RMAPIConfigurationServiceCache.java

```diff
@@ -0,0 +1,55 @@
+package org.folio.config;
+
+import io.vertx.core.Context;
+import org.folio.config.api.RMAPIConfigurationService;
+import org.folio.config.cache.RMAPIConfigurationCache;
+import org.folio.rest.model.OkapiData;
+
+import java.util.concurrent.CompletableFuture;
+
+public class RMAPIConfigurationServiceCache implements RMAPIConfigurationService {
+
+  private RMAPIConfigurationService rmapiConfigurationService;
+
+  public RMAPIConfigurationServiceCache(RMAPIConfigurationService rmapiConfigurationService) {
+    this.rmapiConfigurationService = rmapiConfigurationService;
+  }
+
+  @Override
+  public CompletableFuture<RMAPIConfiguration> retrieveConfiguration(OkapiData okapiData) {
+    RMAPIConfiguration cachedConfiguration = RMAPIConfigurationCache.getInstance()
+      .getValue();
+    if(cachedConfiguration != null){
```

Comment:
one thing that we have to be careful about is if there is a possibility for stale data in cache? I see that if the user updates configuration making a PUT request to mod-kb-ebsco-java, then, yes we are updating cache with latest config data. But there is a possibility of other scenarios like making POST requests directly to configuration module - in case of which our cached data might be stale.

Reply:
It is possible for configuration to become stale if someone sends POST requests to mod-configuration, but cache will expire after several minutes and will be updated.

Reply:
yep, that's true. Hopefully, if we are using stale data too, verification will come back false and we should then be able to make an actual request to get latest!